packages/flutter/test/material/page_transitions_theme_test.dart

```diff
@@ -319,6 +319,81 @@ void main() {
       },
       variant: TargetPlatformVariant.only(TargetPlatform.android),
     );
+
+    testWidgets(
+      'FadeForwardsPageTransitionBuilder does not use ColoredBox for non-opaque routes',
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          MaterialApp(
+            theme: ThemeData(
+              pageTransitionsTheme: const PageTransitionsTheme(
+                builders: <TargetPlatform, PageTransitionsBuilder>{
+                  TargetPlatform.android: FadeForwardsPageTransitionsBuilder(
+                    backgroundColor: Colors.lightGreen,
+                  ),
+                },
+              ),
+            ),
+            home: Builder(
+              builder: (BuildContext context) {
+                return Material(
+                  child: TextButton(
+                    onPressed: () {
+                      Navigator.push(
+                        context,
+                        PageRouteBuilder<void>(
+                          opaque: false,
+                          pageBuilder: (_, _, _) {
+                            return Material(
+                              child: TextButton(
+                                onPressed: () {
+                                  Navigator.push(
+                                    context,
+                                    MaterialPageRoute<void>(builder: (_) => const Text('page b')),
+                                  );
+                                },
+                                child: const Text('push b'),
+                              ),
+                            );
+                          },
+                        ),
+                      );
+                    },
+                    child: const Text('push a'),
+                  ),
+                );
+              },
+            ),
+          ),
+        );
+
+        await tester.tap(find.text('push a'));
+        await tester.pumpAndSettle();
+
+        await tester.tap(find.text('push b'));
+        await tester.pump(const Duration(milliseconds: 400));
```

Comment:
I was just having problems with hardcoded transition values like this. I have a solution in https://github.com/flutter/flutter/pull/171109 but it's not ready yet, so this is probably ok for now.